File: sdk/php/src/Resources/CloudService.php
<?php

namespace HiTechCloud\SDK\Resources;

class CloudService extends BaseResource
{
    /**
     * Shutdown VM
     */
    public function shutdown(int $serviceId, int $vmId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/shutdown');
    }

    /**
     * Reset VM
     */
    public function reset(int $serviceId, int $vmId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/reset');
    }

    /**
     * Set hostname
     */
    public function setHostname(int $serviceId, int $vmId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/hostname', $data);
    }

    /**
     * Get IP pool
     */
    public function getIpPool(int $serviceId, int $vmId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/vms/' . $vmId . '/ippool');
    }

    /**
     * Assign IP
     */
    public function assignIp(int $serviceId, int $vmId, string $pool): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/ippool/' . $pool);
    }

    /**
     * List networks
     */
    public function listNetworks(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/networks');
    }

    /**
     * Create interface
     */
    public function createInterface(int $serviceId, int $vmId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/interfaces', $data);
    }

    /**
     * Get interface
     */
    public function getInterface(int $serviceId, int $vmId, string $iface): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/vms/' . $vmId . '/interfaces/' . $iface);
    }

    /**
     * Update interface
     */
    public function updateInterface(int $serviceId, int $vmId, string $iface, array $data): array
    {
        return $this->http->put('/api/service/' . $serviceId . '/vms/' . $vmId . '/interfaces/' . $iface, $data);
    }

    /**
     * Delete interface
     */
    public function deleteInterface(int $serviceId, int $vmId, string $iface): array
    {
        return $this->http->delete('/api/service/' . $serviceId . '/vms/' . $vmId . '/interfaces/' . $iface);
    }

}

File: sdk/csharp/Resources/CloudService.cs
namespace HiTechCloud.SDK.Resources;

public class CloudService : BaseResource
{
    public CloudService(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ShutdownAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/shutdown", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> ResetAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/reset", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> SetHostnameAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/hostname", id, vmid), data);
    }

    public async Task<Dictionary<string, object?>> IpPoolAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/ippool", id, vmid));
    }

    public async Task<Dictionary<string, object?>> AssignIpAsync(int id, int vmid, string pool)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/ippool/{pool}", id, vmid, pool), null);
    }

    public async Task<Dictionary<string, object?>> NetworksAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/networks", id));
    }

    public async Task<Dictionary<string, object?>> CreateInterfaceAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/interfaces", id, vmid), data);
    }

    public async Task<Dictionary<string, object?>> GetInterfaceAsync(int id, int vmid, string iface)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/interfaces/{iface}", id, vmid, iface));
    }

    public async Task<Dictionary<string, object?>> UpdateInterfaceAsync(int id, int vmid, string iface, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{id}/vms/{vmid}/interfaces/{iface}", id, vmid, iface), data);
    }

    public async Task<Dictionary<string, object?>> DeleteInterfaceAsync(int id, int vmid, string iface)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{id}/vms/{vmid}/interfaces/{iface}", id, vmid, iface));
    }
}
